tuta-sdk/rust/src/date.rs

```diff
@@ -0,0 +1,92 @@
+use std::fmt::Formatter;
+use std::time::{Duration, SystemTime};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use serde::de::{Error, Visitor};
+use uniffi::{FfiConverter, Lift, Lower, MetadataBuffer, RustBuffer};
+
+#[derive(Copy, Clone, PartialEq, PartialOrd, Debug)]
+pub struct Date(SystemTime);
+
+impl Date {
+    pub fn new(time: SystemTime) -> Self {
+        Self(time)
+    }
+
+    pub fn from_millis(millis: u64) -> Self {
+        Self(SystemTime::UNIX_EPOCH + Duration::from_millis(millis))
+    }
+
+    pub fn get_time(self) -> SystemTime {
+        self.0
+    }
+
+    pub fn as_millis(self) -> i64 {
+        self.0.duration_since(SystemTime::UNIX_EPOCH).unwrap().as_millis() as i64
+    }
+}
+
+// Have to do all of this dancing because this isn't quite enough for what we want:
+//
+// uniffi::custom_newtype!(Date, SystemTime);
+//
+// Basically, we want to serialize dates as i64 but let them be passed as native Instant types in FFI
```

Comment:
Would using the `custom_newtype()` macro and using the below implementations for serialize and deserialize not be enough to achieve this?